kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -297,28 +297,36 @@ trait ProcBuilder {
    * @return SPARK_HOME, HIVE_HOME, etc.
    */
   protected def getEngineHome(shortName: String): String = {
+    val homeDirFilter: FilenameFilter = (dir: File, name: String) =>
+      dir.isDirectory && name.contains(s"$shortName-") && !name.contains("-engine")
+
     val homeKey = s"${shortName.toUpperCase}_HOME"
-    val homeVal = env.get(homeKey).orElse {
-      val cwd = Utils.getCodeSourceLocation(getClass).split("kyuubi-server")
-      assert(cwd.length > 1)
-      Option(
-        Paths.get(cwd.head)
-          .resolve("externals")
-          .resolve("kyuubi-download")
-          .resolve("target")
-          .toFile
-          .listFiles(new FilenameFilter {
-            override def accept(dir: File, name: String): Boolean = {
-              dir.isDirectory && name.contains(s"$shortName-")
-            }
-          }))
-        .flatMap(_.headOption)
-        .map(_.getAbsolutePath)
-    }
-    if (homeVal.isEmpty) {
-      throw validateEnv(homeKey)
-    } else {
-      homeVal.get
+    // 1. get from env, e.g. SPARK_HOME, FLINK_HOME
+    env.get(homeKey)
+      .orElse {
+        // 2. get from $KYUUBI_HOME/externals
```

Comment:
this step seems unnecessary

Reply:
I think we still support engine binary bundled mode, e.g. w/o explicit `--spark-provided`, the `build/dist` will bundle the spark binary under `$KYUUBI_HOME/externals`

Reply:
the code path you mentioned is handled already in load-kyuubi-env.sh, we don't need it here

Reply:
make sense.